akka-distributed-data/src/main/scala/akka/cluster/ddata/ORSet.scala

```diff
@@ -241,7 +255,13 @@ final class ORSet[A] private[akka] (
   private[akka] def add(node: UniqueAddress, element: A): ORSet[A] = {
     val newVvector = vvector + node
     val newDot = VersionVector(node, newVvector.versionAt(node))
-    assignAncestor(new ORSet(elementsMap = elementsMap.updated(element, newDot), vvector = newVvector))
+    val newDelta = _delta match {
+      case None ⇒ new ORSet(Map(element → newDot), newDot, deltaOp = AddDeltaOp)
+      case Some(d) ⇒
+        // FIXME handle merge/accumulation of deltas
```

Comment:
yes, `Option` indeed. But I would rather suggest a `List`. The problem with mixed deltas and merging delete deltas is that the removal for `ORSet` is the absence of particular element at certain dot. So to merge two delete deltas we need to keep the latest dot of the delete element (instead of the earliest which is the case in addition). I believe in `ORSet` this is easiest to handle with sending full state. But for other CRDTs, like Removal Wins Add-Remove Set (so mirror of ORSet which is Addition Wins Add-Remove Set) having separate list of deltas to apply might be required, and for the existing `ORSet` would enable easy implementation of removal deltas without full state dissemination.